### DISCUSSION TITLE
[proposal] Maths rendering

### DISCUSSION THREAD

**Discussion by FedeClaudi** [Textualize/rich#409] [Ideas] [answered]
Hi WIll,

Sorry I'm spamming you today, but I wanted to suggest an improvement for rich. 
As far as I can tell there is no easy way to print beautifully formatted latex equations in rich (or in python in general it seems), but there is [unicodeit](https://github.com/svenkreiss/unicodeit) which gets pretty close for simple inline maths expressions.

I was wandering if you'd be interested in integrating it in `rich`, it's very simple to use and the results are quite nice.

For instance:
```python 
from rich import print
from unicodeit import replace

class Math():
    def __init__(self, expression):
        self.expression = replace(expression)

    def __rich_console__(self, *args):
        yield self.expression

print(Math('\\alpha: \\Re \\to \\Im'))
```

gives:
<img width="250" alt="image" src="https://user-images.githubusercontent.com/17436313/97107494-d602bb80-16bf-11eb-9086-8959f2a3b68f.png">

though it might be simpler to just add `replace` to anything being printed: it leaves text unchanged, only tries to replace stuff that looks like a latex symbol. Of course it could be made optional for users that don't like it. 

Let me know what you think of it!

**Comment by willmcgugan** [comment, marked as answer]
I think unicodeit is so simple to use it doesn't need anything added to Rich. Like you said, it's easy to call `replace` as required. And if you prefer wrapping it in a renderable, that's just a few lines.

In general, these things are best left as external libraries. I wouldn't want to add too many dependancies in the main lib. If you wanted to, you could publish a `rich-math` library on Pypi...

**Reply by FedeClaudi** [reply]
that's a great idea!

Thanks
